### PR TITLE
fix: Update CI to use GitHub actions and build for osx, windows, linux

### PR DESCRIPTION
**Pull request by crwilcox**
This PR
1) fixes an issue in DLL discovery for 3.8, 3.9 on windows
2) configures github actions for releasing new wheels
3) removes kokoro based presubmit, continuous, release runs that are now covered here.

### CODE REVIEW

File: src/google_crc32c/__config__.py [hunk 1]
```diff
@@ -27,12 +28,16 @@ def modify_path():
     if path is None:
         return
 
+    # 3.6, 3.7 use path
```
Comment (andrewsg) [inline src/google_crc32c/__config__.py:31]: Duplicate comment with below?

File: tests/conftest.py [hunk 2]
```diff
@@ -13,19 +13,7 @@
...
 
 """py.test shared testing configuration.
-
-This
-
-* Monkey-patches ``mock`` as ``unittest.mock`` for Python 2.7.
 """
 
 import sys
 import unittest
```
Comment (andrewsg) [inline tests/conftest.py:19]: I believe this file can be removed entirely if the mock patch is being removed.

**Review by andrewsg — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 2 above)
Comment (crwilcox) [reply tests/conftest.py:19]: yeah, I can delete it :)

**Comment by crwilcox** [comment]
Will add separate commits to remove some of the old CI that this replaces

**Comment by crwilcox** [comment]
This action won't run until merge, I have a followup PR to update the readme a bit for this project.

**Review by andrewsg — APPROVED** [review]
LGTM, with the caveat that I don't understand the yaml file requirements so if it's hand-crafted I can't review its correctness. Thanks for this work!

**Comment by crwilcox** [comment]
@andrewsg  I did run these with different triggers on my fork so I expect they work as expected 👍

**Pull request merged**